app/assets/javascripts/fae/form/_ajax.js

```diff
@@ -302,9 +307,69 @@ Fae.form.ajax = {
         }
       })
 
-      // stop the event bubbling and running the above toggleClass twice
+      /**
+       * Stop event bubbling and running the above toggleClass twice
+       */
       .on('click', '.boolean :checkbox, .js-checkbox-wrapper :checkbox', function(e){
         e.stopPropagation();
+      })
+
+      /**
+       * Support for a focus state on radio collections
+       */
+      .on('focus blur', '.radio_collection :radio', function(e) {
+        $(this)
+          .closest('.input.radio_collection')
+          .toggleClass('focused', $(this).has(':focus'));
+      })
+
+      /**
+       * Support for a focus state on checkboxes
+       */
+      .on('focus blur', '.boolean :checkbox, .js-checkbox-wrapper :checkbox', function(e) {
+        $(this)
+          .closest('label.boolean, span.checkbox')
+          .toggleClass('focused', $(this).has(':focus'));
+      })
+
+      /**
+       * Support spacebar toggling for checkboxes
+       */
+      .on('keydown', '.boolean, .js-checkbox-wrapper :checkbox', function(e) {
+        if (e.which == 32) {
+          e.preventDefault();
+          $(':checkbox:focus')
+            .closest('label')
+            .trigger('click');
+        }
+      })
+
+      /**
+       * Support for shift+tab off of ms-list element.
+       * By default, $.multiSelect() plugin captures shift+tab and disgards it
+       * @todo This entire method feels very brittle. Possible alternative:
+       * - Create an index of all focusable form elements on page load / DOM mutation
+       * - Use this index to navigate upwards from .ms-list element
+      */
+      .on('keydown', '.ms-list', function(e) {
```

Comment:
Instead of sniffing out the previous input through traversal, could you store an input once it's focused as the last focused input? You'd have to be careful to do this per form (i.e. `$('input').focus(function() { $(this).closest('form').data('last-focused', $(this)); }`) so as to not run afoul of pages with multiple forms such as the AJAX'd ones. 

Once you have that, you could access the last input: `var $prevFocusable = $(this).closest('form').data('last-focused');` and do the rest of the `.focus()` and `.trigger` work.